src/frontends/tensorflow/src/op/non_max_suppression.cpp

```diff
@@ -3,81 +3,154 @@
+        // NonMaxSuppressionV4 has five inputs (same as NonMaxSuppressionV3) and two outputs (selected indices and
+        // valid_outputs) pad_to_max_output_size is a new attribute
+        default_op_checks(node, 5, {"NonMaxSuppressionV4"});
+        iou_threshold = node.get_input(3);
+        score_threshold = node.get_input(4);
+        pad_to_max_output_size = node.get_attribute<bool>("pad_to_max_output_size", false);
+        valid_outputs = true;
+    } else if (op_type == "NonMaxSuppressionV5") {
+        // NonMaxSuppressionV5 has six inputs and three outputs (selected indices, selected scores,
+        // valid_outputs)
+        // soft_nms_sigma is a new input
+        default_op_checks(node, 6, {"NonMaxSuppressionV5"});
+        iou_threshold = node.get_input(3);
+        score_threshold = node.get_input(4);
+        soft_nms_sigma = node.get_input(5);
+        pad_to_max_output_size = node.get_attribute<bool>("pad_to_max_output_size", false);
+        selected_scores = true;
+        valid_outputs = true;
+    } else {
+        TENSORFLOW_OP_VALIDATION(node, false, "Internal error TensorFlow Frontend: no translator for " + op_type);
     }
-    TENSORFLOW_OP_VALIDATION(node, false, "No translator found.");
+
+    auto non_max_suppression = make_shared<NonMaxSuppression>(ov_boxes,
+                                                              ov_scores,
+                                                              max_output_size,
+                                                              iou_threshold,
+                                                              score_threshold,
+                                                              soft_nms_sigma,
+                                                              NonMaxSuppression::BoxEncodingType::CORNER,
+                                                              false,
+                                                              element::i32);
+    auto tf_selected_indices =
+        normalize_selected_indices(non_max_suppression->output(0), max_output_size, pad_to_max_output_size);
+    results.push_back(tf_selected_indices);
+
+    Output<Node> tf_selected_scores;
+    if (selected_scores) {
+        tf_selected_scores =
+            normalize_selected_indices(non_max_suppression->output(1), max_output_size, pad_to_max_output_size);
+        tf_selected_scores = make_shared<Convert>(tf_selected_scores, boxes.get_element_type())->output(0);
+        results.push_back(tf_selected_scores);
+    }
+
+    Output<Node> tf_valid_outputs;
+    if (valid_outputs) {
+        // In TensorFlow valid_outputs output is a scalar
+        tf_valid_outputs = make_shared<Squeeze>(non_max_suppression->output(2))->output(0);
+        results.push_back(tf_valid_outputs);
+    }
+
+    // set output tensor names based on a number of outputs
+    set_node_name(node.get_name(), tf_selected_indices.get_node_shared_ptr());
+    if (valid_outputs && !selected_scores) {
```

Comment:
can we simplify this?
for (size_t i = 0; i < results.size(); ++i) {
    set_node_name(node.get_name() + ":" + std::to_string(i), results[i].get_node_shared_ptr());
    }

Reply:
good simplification. propose to do in the next PR due to CI overloading if no concerns.